src/main/java/sirius/biz/jobs/batch/file/ArchiveImportJob.java

```diff
@@ -0,0 +1,149 @@
+
+import sirius.biz.process.ProcessContext;
+import sirius.biz.process.logs.ProcessLog;
+import sirius.biz.storage.layer1.FileHandle;
+import sirius.biz.storage.layer3.FileParameter;
+import sirius.kernel.commons.Files;
+import sirius.kernel.health.Exceptions;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * Provides an import job which allows to import data from multiple archived files in a specific order.
+ */
+public abstract class ArchiveImportJob extends FileImportJob {
+
+    /**
+     * Contains the file name and input stream of all archive entries whose extensions can be handled.
+     */
+    private final Map<String, InputStream> entries = new HashMap<>();
+
+    /**
+     * Creates a new job for the given process context.
+     *
+     * @param fileParameter the parameter which is used to derive the import file from
+     * @param process       the process context in which the job is executed
+     */
+    protected ArchiveImportJob(FileParameter fileParameter, ProcessContext process) {
+        super(fileParameter, process);
+    }
+
+    @Override
+    protected void executeForSingleFile(FileHandle fileHandle) throws Exception {
+        entries.put(fileHandle.getFile().getName(), fileHandle.getInputStream());
+        importFiles();
+    }
+
+    @Override
+    protected void executeForArchive(FileHandle fileHandle) throws Exception {
+        process.log(ProcessLog.info().withNLSKey("FileImportJob.importingZipFile"));
+
+        try (ZipFile zipFile = new ZipFile(fileHandle.getFile())) {
+            Enumeration<? extends ZipEntry> zipEntries = zipFile.entries();
+
+            int validArchiveEntries = 0;
+
+            while (zipEntries.hasMoreElements()) {
+                ZipEntry zipEntry = zipEntries.nextElement();
+
+                if (!isHiddenFile(zipEntry.getName())
+                    && canHandleFileExtension(Files.getFileExtension(zipEntry.getName()))) {
+                    this.entries.put(zipEntry.getName(), zipFile.getInputStream(zipEntry));
+                    validArchiveEntries++;
+                }
+            }
+
+            if (validArchiveEntries == 0) {
+                throw Exceptions.createHandled().withNLSKey("FileImportJob.noZippedFileFound").handle();
+            }
+
+            importFiles();
+        }
+    }
+
+    /**
+     * Imports data based on the open input streams.
+     *
+     * @throws Exception in case of an exception while importing
+     */
+    protected abstract void importFiles() throws Exception;
+
+    protected List<InputStream> getFiles() {
+        return new ArrayList<>(entries.values());
+    }
+
+    @Nullable
+    protected InputStream getFile(String fileName, boolean isRequired) {
```

Comment:
Why not search the stream on demand here - should greatly simplify the code and male getFile re usable for the same file

Reply:
I'll look into that.